dbms/src/Functions/FunctionFQDN.cpp

```diff
@@ -0,0 +1,56 @@
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <DataTypes/DataTypeString.h>
+#include <Poco/Net/DNS.h>
+#include <Core/Field.h>
+
+
+namespace DB
+{
+
+class FunctionFQDN : public IFunction
+{
+public:
+    static constexpr auto name = "FQDN";
+    static FunctionPtr create(const Context &)
+    {
+        return std::make_shared<FunctionFQDN>();
+    }
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    bool isDeterministic() const override { return false; }
+
+    bool isDeterministicInScopeOfQuery() const override
+    {
+        return false;
+    }
+
+    size_t getNumberOfArguments() const override
+    {
+        return 0;
+    }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & /*arguments*/) const override
+    {
+        return std::make_shared<DataTypeString>();
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers &, size_t result, size_t input_rows_count) override
+    {
+        block.getByPosition(result).column = block.getByPosition(result).type->createColumnConst(
+            input_rows_count, Poco::Net::DNS::thisHost().name())->convertToFullColumnIfConst();
```

Comment:
BTW, we have getFQDNOrHostName function that has two differences:
- it's cached;
- it has fallback to OS hostname if DNS does not work.

Reply:
Right, should it changes to the same behavior as getFQDNOrHostName() ？
That way the function should be named as `getFQDNOrHostName` too.

Reply:
> That way the function should be named as getFQDNOrHostName too.

I think, now it's Ok.